### DISCUSSION TITLE
Uploading a local duckdb database

### DISCUSSION THREAD

**Discussion by slodge** [planetscale/discussion#404] [Q&A] [answered]
Hello

I've got a local 1GB duckdb database.

It has 8 tables and approximately 100 million rows. It's mainly got INT, DATE and VARCHAR fields; it doesn't have any particularly important indicies/keys/stats that I need to preserve in the migration; the inter-table relationship graph is very simple so it's easy to work out a data import order which preserves referential integrity.

It's easy to export this duckdb database to parquet or to csv... However, there doesn't seem to be any "import from file option" in planetscale? 

I'm currently not sure what the best way to migrate this database into planetscale is. Currently the only option I can really think of is to write a script to load and upload the rows using duckdb and odbc mysql drivers - but I think that's slow, inefficient and (in terms of planetscale api) costly... I think it will cost 100 million row writes - which is 2 months of scaler use before I've even started.

Does anyone have any suggestions for better batch import/upload mechanisms?

Thanks

Stuart

**Comment by fooforge** [comment, marked as answer]
Hey @slodge,

I have experimented a bit with DuckDB this morning and, since we do not support `LOAD DATA INFILE` yet, I think, writing a script that reads a CSV and issues INSERT statements for each row is the only option at this time. [This comment on stackoverflow](https://stackoverflow.com/questions/23680395/import-csv-to-mysql-without-load-data-infile) has a memory-efficient example written in Python which uses a prepared insert statement.

If you would like to try this out, feel free to open a ticket with us at https://support.planetscale.com/ and let me know your database name and branch. I will then up the write limit so that you don't have to worry about these writes.

**Comment by slodge** [comment]
Thanks for that - and thanks especially for taking the time to experiment.

I'm fairly new to `duckdb` but have so far found it really rather impressive. I don't think planetscale need to think about duckdb support itself, but in the future some kind of `parquet` upload/insert/append/upsert for individual tables might be handy (all in good time)

I'm currently working in R using dbi, but I can write some Python or other code if necessary... I guess in R I'll need to look at https://search.r-project.org/CRAN/refmans/DBI/html/dbBind.html maybe.... I think this will be needed to insert 100M rows... so leave that with me :)

I'll be in touch with support about this "soon" (it's a hobby project so not immediately!)
